### PR TITLE
new translation.

### PR DESCRIPTION
**Pull request by is52hertz**
### Description/描述
New **language** and new **translation**
> zh_hk & zh_tw

### Related Issues/相关的 issue

 

* fix #ISSUE_NUMBER
* resolve #ISSUE_NUMBER

### Checklist/清单

- [ ] The issues this PR addresses are related to each other/此 PR 解决的 issue 是相关的
- [ ] My changes are all related to the related issue above/我的修改都与上方的相关 issue 有关
- [ ] My changes generate no new warnings/我的修改没有产生新的警告
- [ ] My code builds and runs on my machine/我的代码在我的电脑上可以编译和运行
- [ ] I documented my code/我已经为我的代码添加了注释/文档

### CODE REVIEW

File: src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java [hunk 1]
```diff
@@ -164,14 +164,14 @@ static class Validators extends Validator<Boolean> {
             public Boolean validate(@Nullable ServerCommandSource source, CarpetRule<Boolean> changingRule, Boolean newValue, String userInput) {
                 if (newValue && !Options.redenDebuggerEnabled) {
                     if (source != null) {
-                        source.sendMessage(Text.literal("Warning: ").formatted(Formatting.RED)
-                                .append(Text.literal("The rule "))
-                                .append(Text.literal("redenDebuggerEnabled").styled(style ->
+                        source.sendMessage(Text.translatable("reden.Carpet.warning" + ": ").formatted(Formatting.RED)
```
Comment (zly2006) [inline src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: translation key必须小写

Comment (zly2006) [reply src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: 同时，此处应该使用rich translation（see：https://docs.wispforest.io/owo/rich-translations/）

**Review by zly2006 — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java (hunk 1 above)
Comment (is52hertz) [reply src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: .formatted(Formatting.RED)
这个?

Comment (is52hertz) [reply src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: { "text": "警告", "color": "red"}
这样吗?

Comment (zly2006) [reply src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: > { "text": "警告", "color": "red"} 这样吗?

是的，请修改

File: src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java [hunk 2]
```diff
@@ -164,14 +164,14 @@ static class Validators extends Validator<Boolean> {
             public Boolean validate(@Nullable ServerCommandSource source, CarpetRule<Boolean> changingRule, Boolean newValue, String userInput) {
                 if (newValue && !Options.redenDebuggerEnabled) {
                     if (source != null) {
-                        source.sendMessage(Text.translatable("reden.Carpet.warning" + ": ").formatted(Formatting.RED)
-                                .append(Text.translatable("reden.Carpet.rule"))
-                                .append(Text.translatable("reden.Carpet.redenDebuggerEnabled").styled(style ->
+                        source.sendMessage(Text.translatable("reden.carpet.warning" + ": ").formatted(Formatting.RED)
+                                .append(Text.translatable("reden.carpet.rule"))
+                                .append(Text.translatable("reden.carpet.redendebuggerenabled").styled(style ->
```
Comment (zly2006) [inline src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: 小写不是让你连在一起，下划线。

File: src/main/resources/assets/reden/lang/zh_tw.yml [hunk 3]
```diff
@@ -265,10 +265,10 @@ reden:
       comment: |-
 
   Carpet:
-    warning: 警告
+    warning: { "text": "警告", "color": "red"}
```
Comment (zly2006) [inline src/main/resources/assets/reden/lang/zh_tw.yml]: 再好好看看rich trnaslation格式，本项目中有示例

File: src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java (hunk 2 above)
Comment (zly2006) [reply src/main/java/com/github/zly2006/reden/carpet/RedenCarpetSettings.java]: redenDebuggerEnabled应该用%s方式传入

**Review by zly2006 — CHANGES_REQUESTED** [review]

**Comment by zly2006** [comment]
Link to #29

**Comment by zly2006** [comment]
why ready for review? There are no more commits updated.

**Comment by is52hertz** [comment]
> why ready for review? There are no more commits updated.

I'm going to go ahead and submit it first and create a new PR for a later one, because I'm working on other things right now and can't be bothered to do that.

**Comment by zly2006** [comment]
rich i18n底层代码已更新，see：https://github.com/zly2006/reden-is-what-we-made/blob/ee1ac3c27051e102b50bb0ed01ad9d107d938d38/src/main/java/com/github/zly2006/reden/utils/richTranslation/RichTranslation.kt

**Pull request closed without merging**